### DISCUSSION TITLE
Questions about deleting private Namespace

### DISCUSSION THREAD

**Discussion by klboke** [apolloconfig/apollo#4485] [General]
- As the diagram suggests, when the user deletes the private namespace, the interface is not the same as when the administrator deletes it, and it does not look like it is actually deleted
- Many users are confused when deleting, why am I authorized to delete Namespace, and then prompted to fill in the missing Namespace
- Would it be better to keep the user deletion logic consistent with the administrator deletion logic here?
<img width="981" alt="image" src="https://user-images.githubusercontent.com/18591662/181153242-6b088dc3-5da3-4563-b5c6-a41b59f747d1.png">

**Comment by nobodyiam** [comment]
> Would it be better to keep the user deletion logic consistent with the administrator deletion logic here?

I think we could modify the logic to make it consistent with the administrator deletion logic. But we may add some tips so that the user know what the impact is, e.g. the private namespace in all envs and clusters are deleted, not just the current one.

**Reply by maweibin** [reply]
你好，这个删除namespace的动作在新版本2.4.0版本的时候，和以前完全不一样，现在在某个集群下的namespace删除，会直接删除所有环境下的namespace，但是删除按钮却在当前集群，会给人一种错觉，认为我只是删除当前集群下的namesapce，导致从1.x版本升级到2.4.0的时候，很多用户误删除了其他集群里面正在使用的namespace，或者这个集群下的namespace还是删除当前的集群下的namepsace，应用下的删除namespace是不是放在其他其他

**Comment by klboke** [comment]
Public Namespace deletion also has the same problem
- For example, if a user deletes the Public Namespace of 'Test.kl' and then creates a Private Namespace with the same name, it will report that a Namespace with the same name already exists.
- Of course, continuing to create a Public Namespace with the same name ('Test.kl') will also fail due to the existence of an AppNamespace

- From the user's point of view, this is also not logically expected.
<img width="1791" alt="image" src="https://user-images.githubusercontent.com/18591662/181678924-fb987a3b-7170-4f3c-bf8d-dc51d4b7814f.png">

**Comment by klboke** [comment]
PR：https://github.com/apolloconfig/apollo/pull/4499
